modules/ROOT/pages/tuning-app-design.adoc

```diff
@@ -0,0 +1,252 @@
+
+* `AUTO` +
+Use Automatic to acknowledge a received message automatically only if the flow execution finishes successfully.
+* `MANUAL` +
+Use Manual if you want the app logic to control message acknowledgment.
+* `IMMEDIATE` +
+Use Immediate when you want to acknowledge messages as your app consumes them, before processing them.
+
+Note that if a transaction is opened in the channel, the ACK mode parameter is overruled and the commit handles automatically the message acknowledgment, also the rollback handles automatically the recovery.
+
+
+===== Quality of Service Tab
+You can set the following parameters at the configuration level (Quality of Service tab) with 0 as unlimited. Listener operation can also override the setting. These are the parameters:
+
+* Prefetch Size +
+Defines a prefetch size window. The broker sends as many messages as possible without exceeding the prefetch size window in octets (bytes).
+* Prefetch Count +
+Specifies a global prefetch window in terms of the max number of whole messages. This field may be used in combination with the `prefetch-size` field; a message will only be sent in advance if both prefetch windows allow it.
+
+===== Ack Mode and Prefetch
+
+Acknowledgment mode and QoS prefetch value have a significant effect on consumer throughput. In general, increasing prefetch improves the rate of message delivery to consumers. The Immediate acknowledgment mode yields the best possible rate of delivery. However, in both cases the number of delivered but not yet processed messages will also increase, therefore increasing consumer RAM consumption.
+
+Automatic acknowledgement mode or Manual acknowledgment mode with unlimited prefetch should be used with care. Consumers that consume a lot of messages without acknowledging will lead to memory consumption growth on the broker node they are connected to. Finding a suitable prefetch value is a matter of trial and error and can vary from workload to workload.
+
+==== Publish Parameters
+
+You can set the following parameters at the configuration level (Publisher tab). Publish and Publish-consume operations can also override these parameters. If the use case requires it, you can enable only one of the recommendations:
+
+===== Delivery Mode
+It declares the delivery mode to use when publishing to the AMQP broker. Possible parameters values are:
+
+* `PERSISTENT` +
+Messages marked as persistent and delivered to durable queues are logged to disk on the broker.
+* `TRANSIENT` +
+No logging for the delivered message is performed.
+
+The broker is responsible for the implementation of the recordable storage mechanism for persistent messages. The impact of this feature on performance depends on the AMQP server.
```

Comment:
Shouldn't the first sentence be part of the `PERSISTENT` list item?
And does the second sentence also refer to `PERSISTENT`, or does it apply to both list items?
Clarify by positioning.